src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -199,8 +198,9 @@ internal static int Encode(CoseHeaderMap? map, Span<byte> destination, bool must
 
             if (map._headerParameters.Count == 0 && mustReturnEmptyBstrIfEmpty && !shouldSlipAlgHeader)
             {
-                s_emptyBstrEncoded.CopyTo(destination);
-                return s_emptyBstrEncoded.Length;
+                // Empty Bstr encoded
+                destination[0] = 0x40;
```

Comment:
@Jozkee @bartonjs Could this change the exception type that is thrown? I traced the calls back and it looks like every caller passes a properly-sized buffer to this routine, so I don't expect it would ever throw. But wanted to pick your brains on this to confirm.